aten/src/ATen/mps/MPSDevice.mm

```diff
@@ -20,12 +20,12 @@
   _mtl_device = nil;
 }
 
-MPSDevice::MPSDevice() {
+MPSDevice::MPSDevice():_mtl_device(nil) {
   NSArray* devices = [MTLCopyAllDevices() autorelease];
   for (unsigned long i = 0 ; i < [devices count] ; i++) {
     id<MTLDevice>  device = devices[i];
     if(![device isLowPower]) { // exclude Intel GPUs
-      _mtl_device = device;
+      _mtl_device = [device retain];
```

Comment:
nit: can we convert the assert below into a `TORCH_INTERNAL_ASSERT()` to get it to behave properly?

Reply:
We could not until we move MPS to lazy initialization, otherwise it would be hit every time `torch` is imported on MacOS

Reply:
@malfet , not having `retain` was causing issues. ?